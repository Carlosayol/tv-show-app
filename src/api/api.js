import axios from 'axios';
import { BASE_URL, API_KEY } from '../config/config';

export const getRequest = async (url) => {
  const API_URL = `${BASE_URL}${url}?api_key=${API_KEY}`;
  const response = await axios.get(API_URL)
  return response.data
};