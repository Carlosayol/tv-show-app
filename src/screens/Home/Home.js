import { Dimensions, View, Text, StyleSheet } from 'react-native'
import React from 'react'
import Constants from '../../utils/Constants'
import PopularSeries from '../../components/Popular'
import RecommendedSeries from '../../components/Recommended'

const deviceHeight = Dimensions.get('window').height;

const Home = () => {
  return (
    <View style={styles.container}>
      <PopularSeries />
      <RecommendedSeries />
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Constants.secondaryColor,
    height: deviceHeight,
  },
});

export default Home