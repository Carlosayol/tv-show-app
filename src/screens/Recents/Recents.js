import { View, Text, Dimensions, FlatList, StyleSheet, Image } from "react-native";
import React, { useEffect, useState } from "react";

import Loader from "../../components/Loader";
import { POSTER_IMAGE_URL_B } from "../../config/config";
import Constants from "../../utils/Constants";
import { getRequest } from "../../api/api";
import Rating from "../../components/Rating";

const deviceWidth = Dimensions.get("window").width;

const Recents = (props) => {
  const [loading, setLoading] = useState(true);
  const [recents, setRecents] = useState([]);

  useEffect(() => {
    const getRecents = async () => {
      const data = await getRequest(`/tv/airing_today`);
      setRecents(data.results);
      setLoading(false);
    };

    getRecents();
  }, []);

  return (
    <View style={styles.container}>
      {loading ? (
        <Loader />
      ) : (
        <FlatList
          keyExtractor={(item) => item.id}
          data={recents}
          vertical
          showsVerticalScrollIndicator={false}
          renderItem={(item) => recentCard(item, props)}
        />
      )}
    </View>
  );
};

const recentCard = ({ item }, props) => {
  return (
    <View style={styles.card}>
      <Image source={{ uri: `${POSTER_IMAGE_URL_B}${item.poster_path}` }} style={styles.posterImage} />
      <Text style={styles.heading}>{item.name}</Text>
      <Rating rating={item.vote_average} />
      <View style={{ width: "100%", flexDirection: "row", justifyContent: "flex-end" }}>
        <Text
          style={styles.moreText}
          onPress={() => {
            props.navigation.navigate("RecentDetail", { showId: item.id, poster_path: item.poster_path });
          }}
        >
          Go to view &gt;
        </Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Constants.secondaryColor,
    width: deviceWidth,
    justifyContent: "center",
    alignItems: "center",
  },
  card: {
    width: deviceWidth,
    padding: 40,
    alignItems: "center",
  },
  posterImage: {
    height: 250,
    width: "90%",
    borderRadius: 10,
  },
  textInfo: {
    justifyContent: "space-evenly",
    alignItems: "left",
    padding: 20,
  },
  heading: {
    fontSize: 30,
    fontFamily: "GilroyBold",
    color: Constants.primaryText,
    marginVertical: 20,
  },
  info: {
    color: Constants.tertiaryColor,
    fontFamily: "Gilroy",
  },
  moreText: {
    fontSize: 20,
    fontFamily: "Gilroy",
    color: Constants.primaryColor,
    margin: 20,
  },
});

export default Recents;
