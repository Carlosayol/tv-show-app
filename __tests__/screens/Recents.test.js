import React from "react";
import renderer from "react-test-renderer";
import { render, cleanup, waitFor } from "@testing-library/react-native";

import Home from "../../src/screens/Home";
import { getRequest } from "../../src/api/api";
import Recents from "../../src/screens/Recents";

const mockedNavigate = jest.fn();
jest.mock("@react-navigation/native", () => ({ useNavigation: () => ({ navigate: mockedNavigate }) }));
jest.mock("../../src/api/api");

const createTestProps = (props) => ({
  navigation: {
    navigate: jest.fn(),
  },
  ...props,
});

describe("Recents Screen", () => {
  let props;
  beforeEach(() => {
    props = createTestProps({});
    getRequest.mockResolvedValue({
      results: [],
    });
  });
  afterEach(cleanup);

  it("render base element", () => {
    const tree = renderer.create(<Recents />).toJSON();
    expect(tree.children.length).toBe(1);
  });

  it("renders correctly", () => {
    const tree = renderer.create(<Recents />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
