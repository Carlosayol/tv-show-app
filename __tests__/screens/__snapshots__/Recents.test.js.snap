// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recents Screen renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#191919",
      "flex": 1,
      "justifyContent": "center",
      "width": 750,
    }
  }
>
  <ActivityIndicator
    color="#FFFFFF"
    size="large"
  />
</View>
`;
